File: src/main/java/com/technews/aggregate/releases/springframework/dto/ReleaseResponse.java
package com.technews.aggregate.releases.springframework.dto;

import com.technews.aggregate.releases.springframework.domain.Release;
import lombok.*;

import java.util.List;

@ToString
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ReleaseResponse {
    private String project;
    private String version;
    private String date;
    private String url;
    private List<String> tags;
    private String createdDt;

    public static ReleaseResponse of(Release release) {
        return ReleaseResponse.builder()
                .project(release.getProject())
                .version(release.getVersion())
                .date(release.getDate())
                .url(release.getUrl())
                .tags(release.getTags())
                .createdDt(release.getCreatedDt())
                .build();
    }
}

File: src/main/kotlin/com/technews/aggregate/releases/dto/ReleaseResponse.kt
package com.technews.aggregate.releases.dto

import com.technews.aggregate.releases.domain.Release

data class ReleaseResponse(
    val project: String,
    val version: String,
    val date: String,
    val url: String,
    val tags: List<String>,
    val createdDt: String,
) {
    companion object {
        fun from(release: Release): ReleaseResponse =
            ReleaseResponse(
                project = release.project,
                version = release.version,
                date = release.date,
                url = release.url,
                tags = release.tags,
                createdDt = release.createdDt,
            )
    }
}
